tests/components/sharkiq/test_vacuum.py

```diff
@@ -222,6 +231,40 @@ async def test_device_properties(
     assert getattr(device, device_property) == target_value
 
 
+@pytest.mark.parametrize(
+    ("room_list", "exception"),
+    [
+        (["KITCHEN"], exceptions.HomeAssistantError),
+        (["Kitchen"], None),
+        (["Office"], exceptions.HomeAssistantError),
+        ([], exceptions.HomeAssistantError),
+    ],
+)
+async def test_clean_room_error(
+    hass: HomeAssistant, room_list: list, exception: Exception
+) -> None:
+    """Test clean_room errors."""
+    if exception is None:
```

Comment:
We should avoid if statements in tests. Please create two different tests, one for success and another one where you expect an error.